src/components/@shared/atoms/Table/index.stories.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+import { ComponentStory, ComponentMeta } from '@storybook/react'
+
+import Table, { TableProps } from '@shared/atoms/Table'
+
+export default {
+  title: 'Component/@shared/atoms/Table',
+  component: Table
+} as ComponentMeta<typeof Table>
+
+const Template: ComponentStory<typeof Table> = (args) => <Table {...args} />
+
+interface Props {
+  args: TableProps
+}
+
+export const Default: Props = Template.bind({})
+Default.args = {
+  columns: [
+    {
+      name: 'Column 1 ',
+      maxWidth: '45rem',
+      grow: 1
+    },
+    {
+      name: 'Column 2',
+      maxWidth: '10rem'
+    },
+    {
+      name: 'Column 3',
+      right: true
+    }
+  ],
+  data: []
```

Comment:
would be super handy as a bonus to fill it with some dummy data which we could store in `.jest/__fixtures__/`. Cause the table dependency needs some major version jump and having this isolated in the stories is amazing to make sure to get the update right

Reply:
yes, sure. left it like that because I have to add somehow the `useUserPreferences` hook in here, to get rid of that type error given by the `chainIds` absence, so still on this story

Reply:
right, mocking hooks in storybook... kinda the limit of storybook, it really is meant for pure UI components, could not find any builtin way it has to mock hooks or function calls. Mocking those things for Jest is quite easy though. 

So there are then only 2 possibilities:

1. don't use stories for component, but only test file and either globally or within the test mock `useUserPreferences()` returns
2. refactor component to split-up UI & data fetching in 2 components: one hook-accessing wrapper and one pure markup component which gets all data via props passed down. The UI component can then be tested with Storybook, and the wrapper component tested with Testing Library.

Reply:
quickly tested that last thing mentioned, and wrapped our preview with almost all context providers we have so that should help in some cases to at least get the default returned values. Pushing in a sec

Reply:
not sure but in last CI run, tests succeed. But the last errored storybook build simply says this:

```
ERR! Module not found: Error: Can't resolve '../../../content/networks-metadata.json' in '/home/runner/work/market/market/src/@hooks/useNetworkMetadata'
```

So, you have to have this file generated before running `storybook:build`

Reply:
so you could have `npm run pregenerate` although we only need the last line from the script so far to get this network metadata written out. So maybe `node ./scripts/write-networks-metadata > content/networks-metadata.json` is enough

Reply:
and for the build error you have to merge in the latest `v4` changes as we had contract & subgraph updates in the background

Reply:
> I end up just adding the missing data into the story to be able to render the component

yes, this is the point of mocking. In storybook we never want to have our components try to load actual real data, that would never work

Reply:
had a look into that Table component and basically the `Empty` component it uses should not have any `useUserPreferences()` access. Somebody just hacked this in there instead of constructing the message outside of the component and pass `emptyMessage` to the table component instead. None of our atoms should have any hook access, they are supposed to be pure UI components

Reply:
In this case, should we do the fix in the Table component first?

Reply:
Table component is almost fully covered, only that Empty component lines are missing. Proper way would be to move out any logic around that `useUserPreferences()` access to wherever we actually want this 'No network selected' message, and then from there pass it as `emptyMessage` to component